src/librustc_mir/build/matches/test.rs

```diff
@@ -148,11 +176,26 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                         -> Vec<BasicBlock> {
         let scope_id = self.innermost_scope_id();
         match test.kind {
-            TestKind::Switch { adt_def } => {
+            TestKind::Switch { adt_def, ref variants } => {
                 let num_enum_variants = self.hir.num_variants(adt_def);
-                let target_blocks: Vec<_> =
+                debug!("num_enum_variants: {}", num_enum_variants);
+                debug!("variants.len(): {}", variants.len());
+                debug!("variants: {:?}", variants);
+                let target_blocks: Vec<_> = if variants.into_iter().any(|b| {!b}) {
```

Comment:
no need for the `|b| {!b}`, `|b| !b` would suffice. but maybe `variants.contains(&false)` would be simpler?